_posts/2017-04-04-what-every-developer-should-know-about-ir.md

```diff
@@ -0,0 +1,250 @@
+
+It's important to notice that in pure vector space model, querying is effectively matching the similarity between the query and the documents. Hence, "give me more like this document" -functionality should be trivial to implement: the query is the document.
+
+### TW = TF * IDF
+
+This is the e = mc^2 of information retrieval, and can be written as: term weight equals **term frequency** times **inverse document frequency**.
+
+Most retrieval models choose the assumption that *if a term is common in document but rare in collection, it's really good to distinguish documents from each other within that collection*. So, if a document mentions lots of "dog", it really must be about dogs. It's especially important if "dog" is quite rare term in whole collection. Please note, that this assumption might be wrong as well within collection, but it still is one of the best assumptions we can make when dealing with texts in natural language. Think of it this way: would it make any sense to query with term "dog" to a collection of texts abouts dogs?
+
+TF * IDF weight can be calculated in most simple form for term T in document D belonging to collection C as: T occurrences in D * (1 / T occurrences in C). Usually frequencies are further normalized logarithmically, as we probably want to raise weight of those documents with one or two mentions and diminish the juice on those documents which just repeat the same term over and over, but let's keep things simple.
+
+| id    |  doc                                                                  |
+|-------|-----------------------------------------------------------------------|
+| 1     | A dog eats canned dog food. A Dog do not eat chocolate.               |
+| 2     | Cats eat mice.                                                        |
+| 3     | Cats are usually cute. Dogs too.                                      |
+
+TF for couple of terms in documents.
+
+| term  | TF in documents                            | TF in collection        |
+|-------|----------------------------------------------------------------------|
+| dog   | **1:** 3 **2:** 0 **3:** 1 **4:** 0        | 4                       |
+| cat   | **1:** 0 **2:** 1 **3:** 1 **4:** 0        | 3                       |
+
+By the formula we can calculate that weight for "dog" is 0.75 `= 3 * (1 / 4)` for document 1 and 0.25 `= 1 * (1 / 4)` for document 2. From this we reason that document 1 is more relevant than documents 2 for term "dog" and thus should be raised higher in resultset. Note that weight is not necessarily used to reason if the document is found at all but to order the resultset.
```

Comment:
document 1 is more relevant that document-s- 2 for 
in _the_ result set.
Note that _the_ weight
Lause on monimutkainen, tai ainakin jouduin lukemaan sen monta kertaa. Saisiko jotenkin selkeämmäksi. Note that weight is not necessarily used to reason if the document is found at all but to order the resultset.